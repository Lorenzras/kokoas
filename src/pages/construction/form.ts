import * as Yup from 'yup';

export type BuildingTypeVals =
| '戸建て'
| 'マンション'
| '店舗/事務所'
| 'その他';

export type KeyOfConstructionDetails = keyof typeof initialValues;
export type ConstructionDetailsValues = Partial<Record<KeyOfConstructionDetails, string | number | boolean>>;

/**
 * Set Initial values here in case MUI is shouting about un/controlled components.
 */
export const initialValues = {
  id: '',
  constructionTypeId: '',
  constructionName: '',
  isAgentConfirmed: false,
  cocoConst1: '',
  cocoConst2: '',
  postal: '',
  address1: '',
  address2: '',
  addressKari: '',
  custGroupId: '',
  buildingType: '戸建て' as BuildingTypeVals,
  isChkAddressKari: false,
};

/**
 * Set Validation for fields that requires it.
 * Refer to YUM documentation.
 */
export const validationSchema =  Yup.object(
  {
    custGroupId: Yup
      .string()
      .required('必須です。'),
    postal: Yup
      .string()
      .required('必須です。'),
  } as Partial<Record<KeyOfConstructionDetails, any>>,
);
