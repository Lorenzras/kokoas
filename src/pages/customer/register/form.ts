import * as Yup from 'yup';



export type KeyOfConstructionDetails = keyof ConstructionDetails.SavedData;
export type ConstructionDetailsValues = Partial<Record<KeyOfConstructionDetails, string | number | boolean>>;

enum CustomerFieldNames  {
  'custName',
  'custNameReading',
  'gender',
  'birthYear',
  'birthMonth',
  'birthDay',
  'postal',
  'address1',
  'address2',
  'phone1',
  'phone1Type',
  'phone2',
  'phone2Type',
  'email',
  'isSameAddress',
}

type Customer = {
  [K in keyof typeof CustomerFieldNames ]: string | boolean;
};

export const initialCustomerValue: Customer = {
  custName: '',
  isSameAddress: true,
  custNameReading: '',
  gender: '',
  birthYear: '',
  birthMonth: '',
  birthDay: '',
  postal: '',
  address1: '',
  address2: '',
  phone1: '',
  phone1Type: '',
  phone2: '',
  phone2Type: '',
  email: '',
};

/**
 * Set Initial values here in case MUI is shouting about un/controlled components.
 */
export const initialValues = {
  store: '',
  cocoAG1: '',
  cocoAG2: '',
  yumeAG1: '',
  yumeAG2: '',
  customers: [initialCustomerValue],
};

export type CustomerForm = typeof initialValues;
export type CustomerFormKeys = keyof CustomerForm;
export type  CustomerInstanceKeys = (keyof typeof CustomerFieldNames);

/**
 * Set Validation for fields that requires it.
 * Refer to YUM documentation.
 */
export const validationSchema =  Yup.object(
  {
    custGroupId: Yup
      .string()
      .required('必須です。'),
    postal: Yup
      .string()
      .required('必須です。'),
  } as Partial<Record<KeyOfConstructionDetails, any>>,
);


export const getFieldName = (fieldName : CustomerInstanceKeys) => fieldName;