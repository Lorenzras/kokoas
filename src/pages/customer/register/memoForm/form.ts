import * as Yup from 'yup';

export const initialValues = {
  recordId: '',
  memoType: '',
  contents: '',
  notifyTo: '',
};

export type MemoForm = typeof initialValues;
export type MemoFormKeys = keyof MemoForm;

export const validationSchema =  Yup.object().shape(
  {
    'store' : Yup
      .string()
      .required('必須です。'),
    'cocoAG1' : Yup
      .string()
      .required('必須です。'),
  },
);