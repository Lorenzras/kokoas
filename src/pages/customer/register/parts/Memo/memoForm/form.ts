import * as Yup from 'yup';

export const initialValues = {
  recordId: '',
  memoType: '',
  contents: '',
  notifyTo: '',
};

export type MemoFormType = typeof initialValues;
export type MemoFormKeys = keyof MemoFormType;

export const validationSchema =  Yup.object().shape(
  {
    'store' : Yup
      .string(),
      
    'cocoAG1' : Yup
      .string(),
    
  },
);

export const getFieldName = <T extends MemoFormKeys>(fn : T) : T => fn;