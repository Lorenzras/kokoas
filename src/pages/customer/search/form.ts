import * as Yup from 'yup';

export const customerTypes = [  '全て',  '個人',  '法人' ] as const;
export const recordStatuses = [
  '情報登録のみ',
  '追客中',
  '契約申請中',
  '契約済',
  '工事進行中',
  '工事完了(未精算)',
  '工事完了(精算済)',
] as const;

export type CustomerTypeVals = typeof customerTypes[number];
export type RecordStatus = typeof recordStatuses[number];
export type FormFieldKeys = keyof typeof initialValues;

/**
 * Set Initial values here in case MUI is shouting about un/controlled components.
 */
export const initialValues = {
  custType: '全て' as CustomerTypeVals,
  storeId: '',
  cocoAG: '',
  yumeAG: '',
  cocoConst: '',
  recordStatus: [] as RecordStatus[],
  custName: '',
  contactNum: '',
  email: '',
  address: '',
};

/**
 * Set Validation for fields that requires it.
 * Refer to YUM documentation.
 */
export const validationSchema =  Yup.object(
  {
    custGroupId: Yup
      .string()
      .required('必須です。'),

  },
);
