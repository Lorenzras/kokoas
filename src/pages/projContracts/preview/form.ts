
import * as Yup from 'yup';

export const initialValues = {
  projId: '',
  projName: '',


};

export type TypeOfForm =  typeof initialValues;
export type KeyOfForm = keyof TypeOfForm;

export const getFieldName = (s: KeyOfForm) => s;

export const validationSchema =  Yup.object(
  {
    'projId': Yup
      .string(),
  } as Partial<Record<KeyOfForm, any>>,
);