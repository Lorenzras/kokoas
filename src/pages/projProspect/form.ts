

export const initialValues = {
  projId: '',
  rank: '',
  schedContractPrice: '',
  estatePurchaseDate: '',
  planApplicationDate: '',
  schedContractDate : '',
  memo: '',
};

export type TypeOfForm =  typeof initialValues;
export type KeyOfForm = keyof TypeOfForm;

export const getFieldName = (s: KeyOfForm) => s;