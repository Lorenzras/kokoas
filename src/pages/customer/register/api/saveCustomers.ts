import { APPIDS, KintoneRecord } from '../../../../api/kintone';
import { CustomerForm } from '../form';
import { formToKintCust } from '../helper';

export const saveCustomers = async (formData: CustomerForm) => {
  return KintoneRecord.addRecords({
    app: APPIDS.customers,
    records: formToKintCust(formData),
  }).then(resp => {
    return resp.records;
  });
};


