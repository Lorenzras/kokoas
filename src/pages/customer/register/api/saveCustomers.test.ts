import { saveCustomers } from './saveCustomers';
import { CustomerForm } from '../form';

const testData: CustomerForm =  {
  store: '11',
  cocoAG1: '40',
  cocoAG2: '43',
  yumeAG1: '28',
  yumeAG2: '65',
  customers: [
    {
      key: '',
      custName: 'てすと',
      isSameAddress: true,
      custNameReading: 'テスト',
      gender: '女性',
      birthYear: '1995',
      birthMonth: '6',
      birthDay: '11',
      postal: '441-8124',
      address1: '愛知県豊橋市野依町',
      address2: '',
      phone1: '07014529634',
      phone1Type: '祖父母',
      phone2: '',
      phone2Type: '',
      email: 'lenzras@gmail.com',
      emailType: '',
    },
    {
      key: '',
      custName: 'てすと２',
      isSameAddress: false,
      custNameReading: 'テスト',
      gender: '女性',
      birthYear: '1995',
      birthMonth: '6',
      birthDay: '11',
      postal: '441-8134',
      address1: 'ててててて',
      address2: '',
      phone1: '07014529634',
      phone1Type: '祖父母',
      phone2: '',
      phone2Type: '',
      email: 'lenzras@gmail.com',
      emailType: '',
    },
  ],
} ;

test('Save', async ()=>{
  expect(await saveCustomers(testData)).toMatchSnapshot();
});