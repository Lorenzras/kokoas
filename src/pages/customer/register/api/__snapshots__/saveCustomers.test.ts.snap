// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Save 1`] = `
Array [
  Object {
    "id": "323",
    "revision": "1",
  },
  Object {
    "id": "324",
    "revision": "1",
  },
]
`;
